src/malli/json_schema/parse.cljc

```diff
@@ -0,0 +1,143 @@
+(ns malli.json-schema.parse
+  (:require [malli.core :as m]
+            [malli.util :as mu]
+            [clojure.set :as set]
+            [clojure.string :as str]))
+
+(def annotations #{:title :description :default :examples})
+
+(defn annotations->properties [js-schema]
+  (-> js-schema
+      (select-keys annotations)
+      (set/rename-keys {:examples :json-schema/examples})))
+
+;; Utility Functions
+(defn- map-values
+  ([-fn] (map (fn [[k v]] [k (-fn v)])))
+  ([-fn coll] (sequence (map-values -fn) coll)))
+
+;; Parsing
+(defmulti type->malli :type)
+
+(defn $ref [v]
+  ;; TODO to be improved
+  (keyword (last (str/split v #"/"))))
+
+(defn schema->malli [js-schema]
+  (let [-keys (set (keys js-schema))]
+    (mu/update-properties
+      (cond
+        (-keys :type) (type->malli js-schema)
+
+        (-keys :enum) (into [:enum]
+                            (:enum js-schema))
+
+        (-keys :const) [:enum (:const js-schema)]
+
+        ;; Aggregates
+        (-keys :oneOf) (into
```

Comment:
You can do this by combining and, or, and not.

x XOR y = (Y AND (NOT X)) OR (X AND (NOT Y))

It would be nice if malli just added a xor schema though (which could do the same expansion under the hood).

Reply:
xor != oneOf

https://en.wikipedia.org/wiki/Exclusive_or

> Since it is associative, it may be considered to be an n-ary operator which is true if and only if an odd number of arguments are true. That is, a XOR b XOR ... may be treated as XOR(a,b,...).